pytorch_lightning/loggers/wandb.py

```diff
@@ -262,18 +262,9 @@ def finalize(self, status: str) -> None:
         if self._checkpoint_callback:
             self._scan_and_log_checkpoints(self._checkpoint_callback)
 
-    def _scan_and_log_checkpoints(self, checkpoint_callback: "ReferenceType[ModelCheckpoint]") -> None:
```

Comment:
I propose to split this PR into to parts: 
1) make necessary refactors
2) add feature to mlflow

this makes sure we don't mix a feature for mlflow with changes to the wandb logger. we should be able to separate this cleanly.

Reply:
Should I squash all the commits into these 2 commits, or use this PR for the refactors and open a new one for the mlflow feature?

Reply:
Good question. My suggestion is (how I would do it): 

1. mark this PR as draft for now (top right corner)
2. make a new branch with refactor only, open PR
3. once that PR is merged rebase this one on the changes and continue with the feature here

since we are not in a rush, we can do the refactor first and then the feature. 
Fyi in approx 3 weeks there will be a feature freeze before 1.5 gets released.

Reply:
I created a refactoring PR #9312